mat64/doc.go

```diff
@@ -112,7 +112,34 @@
 // For convenience, a matrix may be used as both a receiver and as an input, i.e.
 //  a.Pow(a, 6)
 //  v.SolveVec(a.T(), v)
-// though in many cases this will cause an allocation. However, mat64 cannot
-// detect arbitrary overlap, and so care should be taken when using matrix views.
+// though in many cases this will cause an allocation (see Element Aliasing).
 // An exception to this rule is Copy, which does not allow a.Copy(a.T()).
+//
+// Element Aliasing
+//
+// Most methods in mat64 modify receiver data. It is forbidden for the modified
+// data region of the receiver to overlap the used data area of the input
+// arguments. The exception to this rule is when the method receiver is equal to one
+// of the input arguments, as in the a.Pow(a, 6) call above, or its implicit transpose.
```

Comment:
I would add something like "mat64 attempts to check for data overlap, for example as constructed with View", possibly in the paragraph saying "don't be clever". Basically, we want to say "mat64 does checking to help you not make mistakes. There are ways to make mistakes using our API, and we try and detect and complain about those. There are lots of ways to create data overlap by circumventing the API. If you need to read this section to see if your overlap is okay, you are being clever. Don't be clever. If you need ... lapack64 ... directly".

In short, I think the current documentation doesn't do enough to explain the goal before telling people not to be clever to get around that goal.